core/Symbols.cc

```diff
@@ -2288,6 +2288,30 @@ uint32_t Method::methodShapeHash(const GlobalState &gs) const {
     return result;
 }
 
+uint32_t Field::fieldShapeHash(const GlobalState &gs) const {
+    uint32_t result = _hash(name.shortName(gs));
+
+    // TODO(jez) type aliases and class aliases participate in the constant resolution fixed
+    // point, which could probably make something start to resolve in a file that doesn't
+    // literally mention the name of this type alias. Add a test for this.
+    // (also maybe double check this logic?)
```

Comment:
Could we add a test like

```ruby
# typed: true

class Parent
  module Inner; end
end

ParentAlias = 1

class A < ParentAlias
  puts(Inner)
end
```

which gets updated to

```ruby
# typed: true

class Parent
  module Inner; end
end

ParentAlias = Parent

class A < ParentAlias
  puts(Inner)
end
```

And assert that it works?

We could even get creative and make it so that `class A` depends on the class alias in a way that the name of `ParentAlias` doesn't appear in the file that has the errors:

```ruby
# typed: true

class Parent
  module Inner; end
end

ParentAlias = 1
# ParentAlias = Parent

Thing = ParentAlias

# -- other.rb --

class A < Thing
  puts(Inner)
end
```

Reply:
(and also delete this comment once such tests are added)